pkg/controller/virtualmachineimage/virtualmachineimage_controller.go

```diff
@@ -126,5 +123,21 @@ func (r *ReconcileVirtualMachineImage) validateVirtualMachineImageSpec() error {
 	if !found {
 		return goerrors.New("storage request in pvc is missing")
 	}
+	if _, err := r.getSource(); err != nil {
+		return err
+	}
 	return nil
 }
+
+func (r *ReconcileVirtualMachineImage) getSource() (string, error) {
+	if r.vmi.Spec.Source.HTTP != "" {
+		if r.vmi.Spec.Source.HostPath != nil {
```

Comment:
I guess instead of nested if statement, having separate if case `if r.vmi.Spec.Source.HTTP != "" &&  r.vmi.Spec.Source.HostPath != nil {}` is better, but it's up to you..